pkg/controller/persistentvolume/types.go

```diff
@@ -92,17 +88,7 @@ func (pvIndex *persistentVolumeOrderedIndex) find(searchPV *api.PersistentVolume
 	//
 	// Searches are performed against a set of access modes, so we can attempt not only the exact matching modes but also
 	// potential matches (the GCEPD example above).
-	allPossibleModes := pvIndex.allPossibleMatchingAccessModes(searchPV.Spec.AccessModes)
-
-	// the searchPV should contain an annotation that allows pre-binding to a claim.
-	// we can use the same annotation value (pvc's namespace/name) and check against
-	// existing volumes to find an exact match. It is possible that a bind is made (ClaimRef persisted to PV)
-	// but the fail to update claim.Spec.VolumeName fails.  This check allows the claim to find the volume
-	// that's already bound to the claim.
-	preboundClaim := ""
-	if createdFor, ok := searchPV.Annotations[createdForKey]; ok {
-		preboundClaim = createdFor
-	}
+	allPossibleModes := pvIndex.allPossibleMatchingAccessModes(claim.Spec.AccessModes)
 
 	for _, modes := range allPossibleModes {
```

Comment:
We discussed this in IM.  This is correct, but we've got a remaining TODO because there will be an easier way to implement this.  Works for now, though.